GLSL preprocessor output preview.

dor666 opened the pull request:
I needed to debug some shaders using preprocessor. It was quite easy to add this functionality to this plugin. Let me know what you think about it.

dor666 committed:
glslValidator

nemoDreamer commented on README.md:
```diff
@@ -39,6 +39,7 @@ Or Settings/Preferences ➔ Packages ➔ Search for `preview`
 - [x] [EmberScript](https://github.com/Glavin001/atom-preview/issues/39)
 - [x] [ng-classify](https://github.com/Glavin001/atom-preview/issues/60)
 - [x] [LiveScript](https://github.com/Glavin001/atom-preview/issues/89)
+- [x] glslPreprocessor
```
I created #145. Could you link to it in the `README`?

nemoDreamer commented on lib/renderer.coffee:
```diff
@@ -284,3 +284,27 @@ module.exports =
+          tmp = require 'tmp'
+          extension = '.' + filepath.split('.').pop();
+          tempFile = tmp.fileSync {prefix: 'atom-preview-editor', postfix: extension};
+          file = fs.writeFileSync tempFile.fd, text
```
Would be sweet if you could use `file`/`writeFile` instead, so as to not block the thread. Thanks to `cb`, all should still work as expected.

nemoDreamer reviewed (changes requested):
Hey @dor666 , great addition, thanks!
Would you mind making two small changes?

nemoDreamer commented:
Oh: one more thing: mind adding a quick little example shader into `spec/samples`? Thanks!

dor666 committed:
Merge branch 'Glavin001'

dor666 committed:
Fixes and refines for glslPreprocessor.

1. Changed temp file operations to async.
2. Added link to glsl issues in README
3. Fixed bug when temp file was not closed by plugin and glslValidator 
could not open it.
4. Added sample fragment shader file.
5. Remove extra blank lines option fixed (multiple lines matched).

dor666 commented:
Hi @nemoDreamer.
I've added the changes you mentioned and fixed additional spotted bugs.
Could you take a look?

nemoDreamer commented on README.md:
```diff
@@ -53,6 +54,10 @@ For instance, if you type or switch tabs,
 how long of a pause before the preview refreshes.
 - `Update On Tab Change` (boolean) -
 Should the preview update to the currently active tab?
+- `glslangValidator` (string) - Path to [glslangValidator](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/) . Needed for [GLSL](https://en.wikipedia.org/wiki/OpenGL_Shading_Language) preview (pre-processed). You can compile it from [source](https://github.com/KhronosGroup/glslang) or download as part of [VulkanSDK](https://www.lunarg.com/vulkan-sdk/).
```
Isn't this gonna read `Glslang Validator`?

nemoDreamer replied:
Also typo: remove the space between the `)` and the `.`

nemoDreamer commented on README.md:
```diff
@@ -53,6 +54,10 @@ For instance, if you type or switch tabs,
+- `glslangValidator` (string) - Path to [glslangValidator](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/) . Needed for [GLSL](https://en.wikipedia.org/wiki/OpenGL_Shading_Language) preview (pre-processed). You can compile it from [source](https://github.com/KhronosGroup/glslang) or download as part of [VulkanSDK](https://www.lunarg.com/vulkan-sdk/).
+- `Remove Extra Blank Lines` (boolean) - Remove extra blank lines in glsl preprocessed output.
+
+removeExtraBlankLines
```
Left-over?

dor666 replied:
Yes, thanks.

nemoDreamer reviewed (approved):
Noice!
(One little comment about the wording in the settings)

nemoDreamer commented on README.md:
```diff
@@ -53,6 +54,10 @@ For instance, if you type or switch tabs,
 - `Update On Tab Change` (boolean) -
 Should the preview update to the currently active tab?
+- `glslangValidator` (string) - Path to [glslangValidator](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/) . Needed for [GLSL](https://en.wikipedia.org/wiki/OpenGL_Shading_Language) preview (pre-processed). You can compile it from [source](https://github.com/KhronosGroup/glslang) or download as part of [VulkanSDK](https://www.lunarg.com/vulkan-sdk/).
+- `Remove Extra Blank Lines` (boolean) - Remove extra blank lines in glsl preprocessed output.
```
And I guess while you're in there, capitalize `glsl`?

dor666 replied:
Yes, correct.

nemoDreamer commented on lib/preview.coffee:
```diff
@@ -12,6 +12,12 @@ module.exports =
     liveUpdate:
       type: 'boolean'
       default: true
+    glslangValidator:
```
Would a different human-readable setting name convert better to title case?
`GLSLValidator -> GLSL Validator`

dor666 committed:
Typos and names consistency in README.

dor666 commented:
Hi, is there still something to polish before it will be merged?

nemoDreamer commented:
Thanks for the addition, @dor666 !
